src/middleware/check-allowed-services-list.ts

```diff
@@ -23,12 +23,23 @@ export const hasHmrcService = async (
   return userServices && findClientInServices(hmrcClientIds, userServices);
 };
 
-export async function checkAllowedServicesList(
+export const hasAllowedRSAServices = async (
+  req: Request,
+  res: Response
+): Promise<boolean> => {
+  const { user } = req.session;
+  const { trace } = res.locals;
+
+  const userServices = await getServices(user.subjectId, trace);
+  return userServices && findClientInServices(rsaAllowList, userServices);
+};
+
+export async function checkRSAAllowedServicesList(
   req: Request,
   res: Response,
   next: NextFunction
 ): Promise<void> {
-  if (await hasHmrcService(req, res)) {
+  if (await hasAllowedRSAServices(req, res)) {
     next();
   } else {
     res.redirect(PATH_DATA.SECURITY.url);
```

Comment:
I forgot to add some logging here, maybe it's useful from a security perspective?